### PR TITLE
HWKAGENT-103 Refactored to use HWKINVENT-180 API

### PR DESCRIPTION
**Pull request by rubenvp8510**
Don't merge yet.

### CODE REVIEW

**Comment by jshaughn** [comment]
Hi Ruben, is this still a work in progress or something that you'd like to have reviewed, even if it's not yet to be merged.

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/storage/AsyncInventoryStorage.java [hunk 1]
```diff
@@ -771,7 +782,7 @@ public void shutdown() {
      * @param tenantId the tenant ID to use, or null which means use the agent tenant ID
      * @return the tenant ID to use
      */
-    private String getTenantIdToUse(String tenantId) {
+    private String  getTenantIdToUse(String tenantId) {
```
Comment (jmazzitelli) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/storage/AsyncInventoryStorage.java]: trivial - delete extra space here

File: hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/storage/AsyncInventoryStorage.java [hunk 2]
```diff
@@ -546,17 +546,28 @@ private void removeResources(List<QueueElement> removeQueueElements) throws Exce
...
+                Collections.reverse(ancestries);
+                for (Resource<?> resource : ancestries) {
+                    String resourceIdPath = "/" + Util.urlEncode(resource.getID().getIDString());
+                    resourcePath.insert(0, resourceIdPath);
                 }
+                CanonicalPath pathPrefix = CanonicalPath.of().tenant(tenandId).feed(feedId).get();
+                CanonicalPath resourceCanonicalPath = CanonicalPath.fromPartiallyUntypedString(resourcePath.toString(),
+                        pathPrefix, org.hawkular.inventory.api.model.Resource.class);
+
+                deleteUrl.append("entity").append(resourceCanonicalPath.toString());
```
Comment (jmazzitelli) [inline hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/storage/AsyncInventoryStorage.java:567]: can you add a small one-line comment here just to give an example of what the final url is going to look like? Just to give someone reading the code an idea of what the inventory REST URL is expected to look like

**Comment by jmazzitelli** [comment]
added "Do No Merge Yet" label as per @rubenvp8510 comment above

**Comment by rubenvp8510** [comment]
@jshaughn 

It is ready for review, but it doesn't pass the tests probably because a bug in 
the Inventory REST API https://github.com/hawkular/hawkular-inventory/pull/268 
it seems like it is fixed now.

**Comment by rubenvp8510** [comment]
This is ready for review and merge if it is ok.

**Comment by jmazzitelli** [comment]
let's wait for inventory to release or at least use a srcdep dependency rather than put 0.17.0.Final-SNAPSHOT into master branch

**Comment by metlos** [comment]
inventory 0.17.0.Final has been released so this can be re-evaluated...

**Comment by jmazzitelli** [comment]
The merge conflict occurs in MonitorService - to fix it, you would take the current master copy of MonitorService as-is and simply replace the one-line change for this PR:
-            url.append("feeds");
-            url.append("entity/feed");

We also need to move to 0.17.0.Final in the pom.xml (this PR currently uses the SNAPSHOT version)

**Comment by jmazzitelli** [comment]
@rubenvp8510 can you rebase this on current master and bump to the inventory release version (see my previous comment on how to fix the conflict)? Once that happens, and we see travis go green, we can merge.

**Pull request merged**